server/service/src/rnr_form/finalise.rs

```diff
@@ -90,3 +102,89 @@ impl From<RepositoryError> for FinaliseRnRFormError {
         FinaliseRnRFormError::DatabaseError(error)
     }
 }
+
+fn create_internal_order(
+    ctx: &ServiceContext,
+    rnr_form: RnRFormRow,
+) -> Result<(), FinaliseRnRFormError> {
+    // Internal Orders are known as requisitions in the code base
+    let requisition_row = RequisitionRow {
+        id: uuid(),
+        user_id: Some(ctx.user_id.clone()),
+        requisition_number: next_number(
+            &ctx.connection,
+            &NumberRowType::RequestRequisition,
+            &ctx.store_id,
+        )?,
+        name_link_id: rnr_form.name_link_id,
+        store_id: rnr_form.store_id,
+        r#type: RequisitionType::Request,
+        status: RequisitionStatus::Sent,
+        created_datetime: Utc::now().naive_utc(),
+        sent_datetime: None,
+        finalised_datetime: None,
+        expected_delivery_date: None,
+        colour: None,
+        comment: Some("Automatically created from R&R Form".to_string()),
+        their_reference: None,
+        max_months_of_stock: 0.0,
+        min_months_of_stock: 0.0,
+        approval_status: None,
+        linked_requisition_id: None,
+        program_id: Some(rnr_form.program_id),
+        period_id: Some(rnr_form.period_id),
+        order_type: None, // Should we capture this in the RnR form?
```

Comment:
We don't have order type in R&R Forms but is required for program requisitions?
Do we want to add to R&R? If we can default when there's only one, it wouldn't be too bad...

Reply:
the order type came from IC requirements, to have 'normal' and 'emergency' orders. In this context, having a 'normal' order makes the most sense.

in theory.. we can link an RNR form to `period` and then to `program_requisition_settings` and then to `program_requisition_order_type` to get the `name` out. however, we'd only want to take non-emergency and we aren't pulling that field through.

might have to add to the R&R form - but yes, defaulting the value if there's only one

Reply:
Just to confirm, are you suggesting we try to look it up here, or we add it to the RNR Form process?

Reply:
the more I think about it, the more i'm leaning to doing a lookup here. though we could equally leave it blank, as I don't see that it is appropriate for the resulting internal order anyway.

Just leave it as `None`.